src/Common/src/TypeSystem/Common/TargetDetails.cs

```diff
@@ -186,5 +186,23 @@ public bool IsWindows
                 return OperatingSystem == TargetOS.Windows;
             }
         }
+
+        /// <summary>
+        /// Maximum number of elements in a HFA type.
+        /// </summary>
+        public int MaximumHfaElementCount
+        {
+            get
+            {
+                // There is a hard limit of 4 elements on an HFA type, see
+                // http://blogs.msdn.com/b/vcblog/archive/2013/07/12/introducing-vector-calling-convention.aspx
+                Debug.Assert(Architecture == TargetArchitecture.ARM ||
+                    Architecture == TargetArchitecture.ARM64 ||
+                    Architecture == TargetArchitecture.X64 ||
```

Comment:
What does this actually mean that it is computed for x86 and x64? As far as I know, we do not use the vector calling convention as the default anywhere - not sure how the link to the vcblog is relevant.